spec/classes/telegraf_spec.rb

```diff
@@ -1,115 +1,122 @@
 require 'spec_helper'
 
 describe 'telegraf' do
-  context 'Supported operating systems' do
-    %w[RedHat CentOS OracleLinux].each do |operatingsystem|
-      [6, 7].each do |releasenum|
-        context "#{operatingsystem} #{releasenum} release specifics" do
-          let(:facts) do
-            {
-              operatingsystem: operatingsystem,
-              operatingsystemrelease: releasenum,
-              operatingsystemmajrelease: releasenum
-            }
-          end
+  on_supported_os(facterversion: '3.6').each do |os, facts|
```

Comment:
There must be a better way to do this. It can be set via an env var so should we fix this in the spec helper? Can we set it via a custom fact?

Reply:
yeah, that'd make sense.  We could set it to a different version depending on whether we're testing against puppet 4, 5 or 6?  IIRC, I've been using 3.6 as that's the version that puppet Inc ships with puppet 4.10.